### PR TITLE
fix(report): update Gitlab template

### LINKED ISSUE #4033: Gitlab template uses deprecated json schema
## Description

The Gitlab report template ([contrib/gitlab.tpl](https://github.com/aquasecurity/trivy/blob/main/contrib/gitlab.tpl)) uses version [14.0.6](https://github.com/aquasecurity/trivy/blob/main/contrib/gitlab.tpl#L3) of the json schema for Gitlab container scanning reports.

This version of the json schema is deprecated. Gitlab suggests to use version 15.0.0 or higher.

image

### PR DESCRIPTION
**Pull request by dklimpel**
## Description

Update Gitlab [Container Scanning](https://docs.gitlab.com/ee/ci/yaml/artifacts_reports.html#artifactsreportscontainer_scanning) template from version 14 to 15, compatible with Gitlab 16.

Discussion #4235

Some information: Gitlab has different reports:

- [Codequality](https://docs.gitlab.com/ee/ci/yaml/artifacts_reports.html#artifactsreportscodequality)
  - [Schema](https://github.com/codeclimate/platform/blob/master/spec/analyzers/SPEC.md#data-types), [Reference in Gitlab docs](https://docs.gitlab.com/ee/ci/testing/code_quality.html#implement-a-custom-tool)
  - The [template in this repo](https://github.com/aquasecurity/trivy/blob/main/contrib/gitlab-codequality.tpl) works with Gitlab 16
 - [Container Scanning](https://docs.gitlab.com/ee/ci/yaml/artifacts_reports.html#artifactsreportscontainer_scanning)
   - [Schema](https://gitlab.com/gitlab-org/security-products/security-report-schemas/-/blob/master/dist/container-scanning-report-format.json)
   - [Template in this repo](https://github.com/aquasecurity/trivy/blob/main/contrib/gitlab.tpl)
   - [Template in Gitlab repo](https://gitlab.com/gitlab-org/security-products/analyzers/container-scanning/-/blob/master/lib/template/trivy.tpl)
   - Both does not work with Gitlab 16.
   - [Diff in schema between Gitlab 14 and 15/16](https://gitlab.com/gitlab-org/security-products/security-report-schemas/-/compare/v14.0.6...v15.0.7?from_project_id=16683102&straight=false#4abff89f3295d75492550cc69ab41b6a6b6eefc2)
 - [Dependency Scanning](https://docs.gitlab.com/ee/ci/yaml/artifacts_reports.html#artifactsreportsdependency_scanning)
   - [Schema](https://gitlab.com/gitlab-org/security-products/security-report-schemas/-/blob/master/dist/dependency-scanning-report-format.json) 
   - Discussed [here / above](https://github.com/aquasecurity/trivy/pull/4035#issuecomment-1553732510) and there is no template at the moment

## Related issues
- Close #4033

## Related PRs
- [x] #4035 

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by knqyf263** [comment]
@dklimpel Thanks for your contribution. Could you take a look at the failure tests?

**Comment by dklimpel** [comment]
> @dklimpel Thanks for your contribution. Could you take a look at the failure tests?

`TestClientServerWithFormat/alpine_3.10_with_gitlab_template` is passing localy.

File: contrib/gitlab.tpl [hunk 1]
```diff
@@ -1,6 +1,29 @@
 {{- /* Template based on https://docs.gitlab.com/ee/user/application_security/container_scanning/#reports-json-format */ -}}
 {
-  "version": "14.0.6",
+  "version": "15.0.7",
+  "scan": {
+    "analyzer": {
+      "id": "trivy",
+      "name": "Trivy",
+      "vendor": {
+        "name": "Aquasecurity"
```
Comment (knqyf263) [inline contrib/gitlab.tpl:9]:
```suggestion
        "name": "Aqua Security"
```

or Aqua_Security if spaces are not allowed.

File: contrib/gitlab.tpl [hunk 2]
```diff
@@ -1,6 +1,29 @@
...
+      },
+      "version": "{{ appVersion }}"
+    },
+    "end_time": "{{ now | date "2006-01-02T15:04:05" }}",
+    "scanner": {
+      "id": "trivy",
+      "name": "Trivy",
+      "url": "https://github.com/aquasecurity/trivy/",
+      "vendor": {
+        "name": "Aquasecurity"
```
Comment (knqyf263) [inline contrib/gitlab.tpl:19]: ditto

**Comment by dklimpel** [comment]
I have seen that I still need to fix this:
`Error: pkg/report/template_test.go:181:45: not enough arguments in call to report.NewTemplateWriter`

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**